lib/src/visitor/async_evaluate.dart

```diff
@@ -2853,6 +2854,31 @@ class _EvaluateVisitor
       throw _exception(error.message, nodeWithSpan.span);
     }
   }
+
+  /// Runs [callback], and converts any [SassRuntimeException]s containing an
+  /// @error to throw a more relevant [SassRuntimeException] with [nodeWithSpan]'s
+  /// source span.
+  T _addErrorSpan<T>(AstNode nodeWithSpan, T callback()) {
+    try {
+      return callback();
+    } on SassRuntimeException catch (error) {
+      if (!error.span.text.startsWith("@error")) rethrow;
+      throw SassRuntimeException(
+          error.message, nodeWithSpan.span, _stackTrace());
+    }
+  }
```

Comment:
I wouldn't add this until it's actually used (or rather, I'd rename `_addErrorSpanAsync()` to `_addErrorSpan()`).

Reply:
Done.